### PR TITLE
Various tidy patches from the symfpu branch

### PR DESCRIPTION
**Pull request by martin-cs**

### CODE REVIEW

File: src/context/cdhashmap.h [hunk 1]
```diff
@@ -192,7 +192,7 @@ class CDOhash_map : public ContextObj {
          bool allocatedInCMM = false) :
     ContextObj(allocatedInCMM, context),
     d_key(key),
-    d_data(),
+    d_data(data),
```
Comment (timothy-king) [inline src/context/cdhashmap.h]: This makes me nervous. The initialization order in this function has been very picky. I'd like to see larger scale tests on just changing this line. (paranoid.)

File: src/context/cdhashmap.h [hunk 2]
```diff
@@ -410,14 +410,29 @@ class CDHashMap : public ContextObj {
...
     } else {
       (*i).second->set(d);
       return false;
     }
   }
 
+  // Use this for pointer data d allocated in context memory at this
+  // level.  THIS IS HIGHLY EXPERIMENTAL.  It seems to work if ALL
+  // your data objects are allocated from context memory.
+  void insertDataFromContextMemory(const Key& k, const Data& d) {
```
Comment (timothy-king) [inline src/context/cdhashmap.h]: If we do not need this, I would avoid adding it.

File: test/unit/Makefile.am [hunk 3]
```diff
@@ -34,8 +34,8 @@ UNIT_TESTS += \
 	context/cdo_black \
 	context/cdlist_black \
 	context/cdlist_context_memory_black \
-	context/cdmap_black \
-	context/cdmap_white \
+	context/cdhashmap_black \
```
Comment (timothy-king) [inline test/unit/Makefile.am]: Are these changes necessary?

**Comment by martin-cs** [comment]
Thanks for the comments Tim.  That part is a patch from Morgan and TBH I have no idea.  Having a chat with @barrettcw and team, they suggest they will take over that patch.  So I'm going to close this and create a new patch set without it.
